### PR TITLE
Flesh out sample project

### PR DESCRIPTION
**Pull request by ZacSweers**
This fleshes out the sample project and adds a number of opportunistic fixes and tweaks to support that setup. Now the sample has a few different cases covered:
- Multiple build types and variants
- Multiple dependencies with varying levels of implementation/api transitives
- Now serves as a representative example integration

### CODE REVIEW

**Comment by ZacSweers** [comment]
Well this is weird, the compiled `c` sample library jar is empty 🤔 

image

**Comment by ZacSweers** [comment]
Wow, sourceSets were necessary for c86dfd8

**Comment by dsvoronin** [comment]
> Well this is weird, the compiled `c` sample library jar is empty 🤔
> 
> image

could it be the same issue i faced?
https://github.com/avito-tech/avito-android/pull/153/files

keeper couldn't see call from library

**Comment by ZacSweers** [comment]
@dsvoronin it could be, but let's not hijack this PR discussion :)

File: .github/workflows/run_instrumentation_tests.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+function printthreads {
+  echo "Thread dump"
+  jps|grep -E 'KotlinCompileDaemon|GradleDaemon'| awk '{print $1}'| xargs jstack
+}
+
+trap printthreads SIGINT
+
+echo "Running with Proguard!"
+timeout --signal=SIGINT 5m ./gradlew connectedExternalStagingAndroidTest -Pandroid.enableR8=false --stacktrace -PkeeperTest.agpVersion=${CI_AGP_VERSION}
```
Comment (jschear) [inline .github/workflows/run_instrumentation_tests.sh]: Any reason to have a timeout for proguard but not r8?

File: keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/KeeperExtension.kt [hunk 2]
```diff
@@ -57,6 +57,12 @@ open class KeeperExtension @Inject constructor(objects: ObjectFactory) {
    * Example: `listOf("-Xdebug", "-Xrunjdwp:transport=dt_socket,address=5005,server=y,suspend=y")`
    */
   val r8JvmArgs: ListProperty<String> = objects.listProperty()
+
+  /** Emit extra debug information, useful for bug reporting. */
+  var emitDebugInformation: Property<Boolean> = objects.property<Boolean>().apply {
```
Comment (jschear) [inline keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/KeeperExtension.kt]: nit: I think `convention` returns the property so you can get rid of `apply`

File: .github/workflows/run_instrumentation_tests.sh (hunk 1 above)
Comment (ZacSweers) [reply .github/workflows/run_instrumentation_tests.sh]: Leftover from when I was seeing hangups on CI, let me just remove it all together

File: keeper-gradle-plugin/build.gradle.kts [hunk 3]
```diff
@@ -48,6 +48,10 @@ tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach
     }
 }
 
+sourceSets {
+    findByName("test")?.resources?.srcDirs("$buildDir/pluginUnderTestMetadata")
```
Comment (jschear) [inline keeper-gradle-plugin/build.gradle.kts]: nit: can we use `getByName` and fail fast if this isn't working?

Comment (ZacSweers) [reply keeper-gradle-plugin/build.gradle.kts]: daf340b

File: keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/VariantClasspathJar.kt [hunk 4]
```diff
@@ -28,36 +34,100 @@ import kotlin.system.measureTimeMillis
...
     measureTimeMillis {
       project.logger.debug("$LOG: Diffing androidTest jars and app jars")
-      val appJars = appRuntime.filter { it.extension == "jar" }.mapTo(mutableSetOf()) { it.nameWithoutExtension }
-      val distinctAndroidTestJars = androidTestRuntime.filter {
-        it.extension == "jar" && it.nameWithoutExtension !in appJars
+      val appJars = appConfiguration.artifactView().files.filterTo(LinkedHashSet()) { it.extension == "jar" }
+      val debug = emitDebugInfo.get()
+      if (debug) {
+        project.file("${project.buildDir}/${KeeperPlugin.INTERMEDIATES_DIR}/${archiveFile.get().asFile.nameWithoutExtension}AppJars.txt").apply {
+          writeText(appJars.sortedBy { it.path }
```
Comment (jschear) [inline keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/VariantClasspathJar.kt]: Is it worth factoring out this block? The three usages in this function look pretty similar to me.

File: keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/KeeperExtension.kt (hunk 2 above)
Comment (ZacSweers) [reply keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/KeeperExtension.kt]: 28097e1

File: keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/VariantClasspathJar.kt (hunk 4 above)
Comment (ZacSweers) [reply keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/VariantClasspathJar.kt]: c884e29

**Pull request merged**